### PR TITLE
[typescript/en] Add template literal types

### PR DESCRIPTION
**Pull request by edazpotato**
- [x] I solemnly swear that this is all original content of which I am the original author
- [x] Pull request title is prepended with `[language/lang-code]` (example `[python/fr-fr]` or `[java/en]`)
- [x] Pull request touches only one file (or a set of logically related files with similar changes made)
- [x] Content changes are aimed at *intermediate to experienced programmers* (this is a poor format for explaining fundamental programming concepts)

### CODE REVIEW

File: typescript.html.markdown [hunk 1]
```diff
@@ -237,6 +237,16 @@ if (state.type === "success") {
   console.error(state.message);
 }
 
+// Template Literal Types
+// Use to create complex stirng types
```
Comment (Menelion) [inline typescript.html.markdown:241]:
```suggestion
// Use to create complex string types
```

**Review by Menelion — CHANGES_REQUESTED** [review]
Thanks! Please fix the typo (you can just commit suggestion) and I'll merge.

**Comment by edazpotato** [comment]
Done!

**Review by Menelion — APPROVED** [review]
Thank you!

**Pull request merged**